lib/msf/core/post/file.rb

```diff
@@ -73,8 +73,14 @@ def dir(directory)
       return session.fs.dir.entries(directory)
     end
 
+    if session.type == 'powershell'
+      dir = session.shell_command_token("Get-ChildItem \"#{directory}\" | Format-Table Name").split(/[\r\n]+/)
+      dir.slice!(0..2)
```

Comment:
There's a paranoid part of me that wants to check the size before we slice.

Reply:
Cool, will add a check!